crates/router/src/connector/shift4/transformers.rs

```diff
@@ -290,13 +408,73 @@ pub enum NextAction {
     None,
 }
 
-impl<F, T>
-    TryFrom<types::ResponseRouterData<F, Shift4PaymentsResponse, T, types::PaymentsResponseData>>
+#[derive(Debug, Serialize, Deserialize)]
+pub struct NuveiCardToken {
+    pub id: String,
+}
+
+impl<F>
+    TryFrom<
+        types::ResponseRouterData<
+            F,
+            Shift4ThreeDsResponse,
+            types::PaymentsAuthorizeData,
+            types::PaymentsResponseData,
+        >,
+    > for types::RouterData<F, types::PaymentsAuthorizeData, types::PaymentsResponseData>
+{
+    type Error = Error;
+    fn try_from(
+        item: types::ResponseRouterData<
+            F,
+            Shift4ThreeDsResponse,
+            types::PaymentsAuthorizeData,
+            types::PaymentsResponseData,
+        >,
+    ) -> Result<Self, Self::Error> {
+        let redirection_data = item
+            .response
+            .redirect_url
+            .map(|url| services::RedirectForm::from((url, services::Method::Get)));
+        Ok(Self {
+            status: if redirection_data.is_some() {
+                enums::AttemptStatus::AuthenticationPending
+            } else {
+                enums::AttemptStatus::Pending
+            },
+            request: types::PaymentsAuthorizeData {
+                enrolled_for_3ds: item.response.enrolled,
+                ..item.data.request
+            },
+            response: Ok(types::PaymentsResponseData::TransactionResponse {
+                resource_id: types::ResponseId::NoResponseId,
```

Comment:
How can we track this payment without any id?

Reply:
After the `/complete` call the status of the transaction and connector_transaction_id will be updated.